### ISSUE TITLE
SOAP error in set-network-adapter with VBox 4.1.2

### ISSUE THREAD

**Issue by coopernurse** [tbatchelli/vmfest#7]
Hi,

Cool project.  I'm just getting started, and tried executing:

```
user=> (def my-machine (create-machine my-server "my-machine" "Ubuntu_64" basic-config-me 
    "/Users/james/oracle/virtualbox/ubuntu-10-10-64bit-server.vdi"))
INFO  virtualbox - create-machine: Creating machine my-machine in /Users/james/.vmfest/nodes/my-machine/my-machine.vbox,  overwriting previous contents
javax.xml.ws.soap.SOAPFaultException: Method 'ns2:INetworkAdapter_attachToBridgedInterface' not implemented: method name or namespace not recognized (NO_SOURCE_FILE:35)
```

Looking at the vmfest source, I think this is coming from:  set-network-adapter (src/vmfest/virtualbox/machine.clj line 312)

My guess is that VBox 4.1 changed the SOAP interface from 4.0.  I'll look at their documentation and see if I can figure it out, but if this is something I'm doing wrong please let me know.

cheers

-- James

**Comment by coopernurse** [comment]
Ah, as I look more closely I see you're pulling in the vboxjws 4.0.2 client JAR.  Looks like Oracle releases Java bindings for the SOAP server.  Ok.  

I wonder how vmfest should handle this -- different folks will have different versions of vbox installed, and presumably vmfest would want to be able to delegate to the correct SOAP version.

I'll fork vmfest and see if I can figure out how to get the 4.1 bindings working locally.

**Comment by tbatchelli** [comment]
Multi-version support is a goal, but not a top priority at this stage. Something that I have in the back of my brain. There is also the need to support both local bindings and SOAP, and that is another bag of hurt.

**Comment by coopernurse** [comment]
Ok, cool.  I got parts of this working locally against vbox 4.1.  It looks like the pallet.clj that's part of pallet needs to change too (the catch block references the 4.0 SOAP bindings).

**Comment by tbatchelli** [comment]
good catch. I am now updating pallet.compute.vmfest so I will update my branch of vmfest to not leak virtualbox specific exceptions. For now, just updating pallet to reference the 4.1 exceptions should get you going.

**Comment by tbatchelli** [comment]
Fixed in `develop` by ebf865b15b174dc195f56fc38d95d819227b310e
